### PR TITLE
Pv feature2

### LINKED ISSUE #9113: BUG: replace int64 dtype with bool coerces to int64
```
# NG
s = pd.Series([1, 2, 3])
s.replace(2, True)
#0    1
#1    1
#2    3
# dtype: int64
```

If target has `object` dtype, it works.

```
s = pd.Series(['x', 2, 3])
s.replace(2, True)
#0       x
#1    True
#2       3
# dtype: object
```

### PR DESCRIPTION
**Pull request by pv8473h12**
- [x] closes #9113
- [x] tests added / passed
- [x] passes `black pandas`
- [x] new test added: test_replace_no_cast()

### CODE REVIEW

File: pandas/tests/series/test_replace.py [hunk 1]
```diff
@@ -305,3 +305,16 @@ def test_replace_with_no_overflowerror(self):
...
+    @pytest.mark.parametrize(
+        "ser, to_replace, value, exp",
+        [([1, 2, 3], 2, True, [1, True, 3]), (["x", 2, 3], 2, True, ["x", True, 3])],
+    )
+    def test_replace_no_cast(self, ser, to_replace, value, exp):
+        # GH 9113
+        # BUG: replace int64 dtype with bool coerces to int64
+
+        series = pd.Series(ser)
+        result = series.replace(to_replace, value)
```
Comment (jreback) [inline pandas/tests/series/test_replace.py]: to_replace and value are the same for the cases, so don't make these args, just put it in the code

**Review by jreback — CHANGES_REQUESTED** [review]

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm. ping on green.

**Comment by pv8473h12** [comment]
All green @jreback

**Comment by WillAyd** [comment]
Thanks @pv8473h12

**Pull request merged**